metadata-ingestion/src/datahub/ingestion/source/sql/redshift.py

```diff
@@ -691,10 +691,11 @@ def _get_sources_from_query(self, db_name: str, query: str) -> List[LineageDatas
         return sources
 
     def get_db_name(self, inspector: Optional[Inspector] = None) -> str:
-        db_name = getattr(self.config, "database")
-        db_alias = getattr(self.config, "database_alias")
+        db_name = self.config.database
+        db_alias = self.config.database_alias
         if db_alias:
             db_name = db_alias
+        assert db_name
```

Comment:
Is this guaranteed to exist? The postgres config states:
```
    database: Optional[str] = Field(
        default=None,
        description="database (catalog). If set to Null, all databases will be considered for ingestion.",
    )
```
so None seems like a valid value

Reply:
we create the urns from this and the return type is annotated as string

if we actually returned `None` here, the urns would look like `urn:dataset(snowflake,None.schema.table,PROD)` which is definitely not correct. As such, I think this one is ok

fyi @treff7es